src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1672,11 +1671,10 @@ private unsafe string InternalSubString(int startIndex, int length)
 
             string result = FastAllocateString(length);
 
-            fixed (char* dest = &result._firstChar)
-            fixed (char* src = &_firstChar)
-            {
-                wstrcpy(dest, src + startIndex, length);
-            }
+            Buffer.Memmove(
+                elementCount: (uint)result.Length, // derefing Length now allows JIT to prove 'result' not null below
```

Comment:
>// derefing Length now allows JIT to prove 'result' not null below

The `string` above isn't marked nullable, isn't that already 'proven'?

Reply:
nullable annotations are not an enforced contract. The JIT cannot use them for optimizations.